In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import plotly.express as px

In [ ]:
df=pd.read_csv('/content/WHRData2021.csv')

In [ ]:
df

Country name  Population 2020  Population 2019  \
0        United States      331002647.0      328239523.0   
1                Egypt      102334403.0      100388073.0   
2              Morocco       36910558.0       36471769.0   
3              Lebanon        6825442.0        6855713.0   
4         Saudi Arabia       34813867.0       34268528.0   
..                 ...              ...              ...   
161              Yemen       29825968.0       29161922.0   
162             Kosovo        1932774.0        1794248.0   
163  Somaliland region              NaN              NaN   
164       North Cyprus              NaN              NaN   
165        South Sudan       11193729.0       11062113.0   

     COVID-19 deaths per 100,000 population in 2020  Median age  Island  \
0                                           104.451        38.3       0   
1                                             7.457        25.3       0   
2                                            20.016        29.6       0   
3                                            21.508        31.1       0   
4                                            17.875        31.9       0   
..                                              ...         ...     ...   
161                                           2.045        20.3       0   
162                                          68.916        35.0       0   
163                                             NaN         NaN       0   
164                                             NaN         NaN       0   
165                                           0.563        19.2       0   

     Index of exposure to COVID-19  infections in other countries as of March 31  \
0                                                1.688                             
1                                                1.627                             
2                                                2.336                             
3                                                1.891                             
4                                                1.250                             
..                                                 ...                             
161                                              1.005                             
162                                              3.134                             
163                                                NaN                             
164                                                NaN                             
165                                              1.076                             

     Log of average distance to SARS countries  WHO Western Pacific Region  \
0                                        9.315                           0   
1                                        9.007                           0   
2                                        9.226                           0   
3                                        8.956                           0   
4                                        8.897                           0   
..                                         ...                         ...   
161                                      8.966                           0   
162                                      9.015                           0   
163                                        NaN                           0   
164                                        NaN                           0   
165                                      9.066                           0   

     Female head of government  Index of institutional trust  \
0                            0                         0.250   
1                            0                         0.446   
2                            0                         0.397   
3                            0                         0.107   
4                            0                         0.651   
..                         ...                           .

In [ ]:
df.shape


(166, 17)

the shape of data is 166 rows and 17 columns

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 166 entries, 0 to 165
Data columns (total 17 columns):
 #   Column                                                                       Non-Null Count  Dtype  
---  ------                                                                       --------------  -----  
 0   Country name                                                                 166 non-null    object 
 1   Population 2020                                                              164 non-null    float64
 2   Population 2019                                                              164 non-null    float64
 3   COVID-19 deaths per 100,000 population in 2020                               163 non-null    float64
 4   Median age                                                                   163 non-null    float64
 5   Island                                                                       166 non-null    int64  
 6   Index of exposure to COVID-19  infections 

In [ ]:
df.isna().sum()

Country name                                                                     0
Population 2020                                                                  2
Population 2019                                                                  2
COVID-19 deaths per 100,000 population in 2020                                   3
Median age                                                                       3
Island                                                                           0
Index of exposure to COVID-19  infections in other countries as of March 31      2
Log of average distance to SARS countries                                        2
WHO Western Pacific Region                                                       0
Female head of government                                                        0
Index of institutional trust                                                     1
Gini coefficient of income                                                       2
All-

In [ ]:
df1=df.drop(['All-cause death count, 2017','All-cause death count, 2018','All-cause death count, 2019','Excess deaths in 2020 per 100,000 population, relative to 2017-2019 average','All-cause death count, 2020'
         ], axis=1)

df1.dropna(inplace=True)

In [ ]:
df1

Country name  Population 2020  Population 2019  \
0    United States      331002647.0      328239523.0   
1            Egypt      102334403.0      100388073.0   
2          Morocco       36910558.0       36471769.0   
3          Lebanon        6825442.0        6855713.0   
4     Saudi Arabia       34813867.0       34268528.0   
..             ...              ...              ...   
159        Uruguay        3473727.0        3461734.0   
160     Uzbekistan       33469199.0       33580650.0   
161          Yemen       29825968.0       29161922.0   
162         Kosovo        1932774.0        1794248.0   
165    South Sudan       11193729.0       11062113.0   

     COVID-19 deaths per 100,000 population in 2020  Median age  Island  \
0                                           104.451        38.3       0   
1                                             7.457        25.3       0   
2                                            20.016        29.6       0   
3                                            21.508        31.1       0   
4                                            17.875        31.9       0   
..                                              ...         ...     ...   
159                                           5.210        35.6       0   
160                                           1.834        28.2       0   
161                                           2.045        20.3       0   
162                                          68.916        35.0       0   
165                                           0.563        19.2       0   

     Index of exposure to COVID-19  infections in other countries as of March 31  \
0                                                1.688                             
1                                                1.627                             
2                                                2.336                             
3                                                1.891                             
4                                                1.250                             
..                                                 ...                             
159                                              0.572                             
160                                              1.141                             
161                                              1.005                             
162                                              3.134                             
165                                              1.076                             

     Log of average distance to SARS countries  WHO Western Pacific Region  \
0                                        9.315                           0   
1                                        9.007                           0   
2                                        9.226                           0   
3                                        8.956                           0   
4                                        8.897                           0   
..                                         ...                         ...   
159                                      9.711                           0   
160                                      8.620                           0   
161                                      8.966                           0   
162                                      9.015                           0   
165                                      9.066                           0   

     Female head of government  Index of institutional trust  \
0                            0                         0.250   
1                            0                         0.446   
2                            0                         0.397   
3                            0                         0.107   
4                            0                         0.651   
..                         ...                           ...   
159                          0        

In [ ]:
df1.isna().sum()

Country name                                                                   0
Population 2020                                                                0
Population 2019                                                                0
COVID-19 deaths per 100,000 population in 2020                                 0
Median age                                                                     0
Island                                                                         0
Index of exposure to COVID-19  infections in other countries as of March 31    0
Log of average distance to SARS countries                                      0
WHO Western Pacific Region                                                     0
Female head of government                                                      0
Index of institutional trust                                                   0
Gini coefficient of income                                                     0
dtype: int64

here we remove all null values 

In [ ]:
fig=px.scatter(df1,x='Population 2020',y='Median age',color='Country name',title='Population of countries with its median age')
fig.show()

In [ ]:
df2=df1.loc [:,['Country name','COVID-19 deaths per 100,000 population in 2020']]

In [ ]:
df3=df2.sort_values(ascending=False,by='COVID-19 deaths per 100,000 population in 2020').head(10)

In [ ]:
df3

Country name  COVID-19 deaths per 100,000 population in 2020
15                  Belgium                                         168.496
146                Slovenia                                         129.730
90   Bosnia and Herzegovina                                         123.445
17                    Italy                                         122.654
127         North Macedonia                                         120.141
141                    Peru                                         114.279
91                 Bulgaria                                         109.032
16                    Spain                                         108.731
133              Montenegro                                         108.588
11           United Kingdom                                         108.450

In [ ]:
fig=px.bar(df3,x='Country name',y='COVID-19 deaths per 100,000 population in 2020',title='Top 10 countries with the most deaths per 100,000 population')
fig.show()

first place belgium then slovenia etc...

In [ ]:
df4=df.loc[:,['Country name','Population 2020','Population 2019']].sort_values(ascending=False,by='Population 2020').head(10)

In [ ]:
df4

Country name  Population 2020  Population 2019
29          China     1.439324e+09     1.397715e+09
30          India     1.380004e+09     1.366418e+09
0   United States     3.310026e+08     3.282395e+08
9       Indonesia     2.735236e+08     2.706256e+08
8        Pakistan     2.208923e+08     2.165653e+08
32         Brazil     2.125594e+08     2.110495e+08
34        Nigeria     2.061396e+08     2.009636e+08
10     Bangladesh     1.646894e+08     1.630462e+08
73         Russia     1.459345e+08     1.443735e+08
33         Mexico     1.289328e+08     1.275755e+08

In [ ]:
fig=px.bar(df4,x='Country name',y=['Population 2020','Population 2019'],barmode='group',title='Top 10 countries with highest population in 2020')

fig.show()

we know that China and India are top of this segment

In [ ]:
df5=df1.loc[:,['Country name','Median age','Population 2020','COVID-19 deaths per 100,000 population in 2020']]
df5

Country name  Median age  Population 2020  \
0    United States        38.3      331002647.0   
1            Egypt        25.3      102334403.0   
2          Morocco        29.6       36910558.0   
3          Lebanon        31.1        6825442.0   
4     Saudi Arabia        31.9       34813867.0   
..             ...         ...              ...   
159        Uruguay        35.6        3473727.0   
160     Uzbekistan        28.2       33469199.0   
161          Yemen        20.3       29825968.0   
162         Kosovo        35.0        1932774.0   
165    South Sudan        19.2       11193729.0   

     COVID-19 deaths per 100,000 population in 2020  
0                                           104.451  
1                                             7.457  
2                                            20.016  
3                                            21.508  
4                                            17.875  
..                                              ...  
159                                           5.210  
160                                           1.834  
161                                           2.045  
162                                          68.916  
165                                           0.563  

[163 rows x 4 columns]

In [ ]:
fig=px.scatter(df5,x='Median age',y='COVID-19 deaths per 100,000 population in 2020',color='Country name',size='Population 2020',size_max=50,title='Median age of countries VS Covid19 deaths per 100,000')
fig.show()

In [ ]:
fig=px.histogram(df1,x='Index of exposure to COVID-19  infections in other countries as of March 31',color='Country name',title='covid19 infections among countries')
fig.show()

In [ ]:
top_10=df1.loc[:,['Country name','Index of institutional trust']].sort_values(ascending=False,by='Index of institutional trust').head(10)

In [ ]:
top_10

Country name  Index of institutional trust
62                 Rwanda                         0.906
27              Singapore                         0.869
101                  Cuba                         0.729
88                 Bhutan                         0.722
52                   Laos                         0.708
158  United Arab Emirates                         0.694
143                 Qatar                         0.676
151           Switzerland                         0.660
4            Saudi Arabia                         0.651
36               Tanzania                         0.621

In [ ]:
fig=px.bar(top_10,x='Country name',y='Index of institutional trust',title='Top10 countries with index of institutional trust')
fig.show()

top 10 coutries list follows:rwanda, singapore etc....